TWLight/applications/models.py

```diff
@@ -319,16 +318,24 @@ def update_app_status_on_save(sender, instance, **kwargs):
             instance.date_closed = localtime(now()).date()
             instance.days_open = 0
 
-# Authorize editor to access resource after an application is saved as approved
-# or sent. We need to use post_revision_commit from django-reversion to get the
-# **current** revision user. Just listening to the post_save signal will grab
-# the user from the previously saved version.
+# Authorize editor to access resource after an application is saved as sent.
 @receiver(post_save, sender=Application)
 def post_revision_commit(sender, instance, **kwargs):
-    if reversion.is_active() and (instance.status == 2 or instance.status == 4) and not instance.imported:
+    # Check if an authorization already exists.
+    if instance.specific_stream:
+        existing_authorization = Authorization.objects.filter(
```

Comment:
It doesn't look like we're actually using this anywhere?

Reply:
Yep, great catch, I wrote the check but forgot to actually use it. This prompted me to give a bit more thought to how authorizations work in relation to renewals. We weren't really handling that properly before, so I've added some logic for updating an existing authorization if one exists and a related test.

Reply:
It might seem a little superfluous right now to edit the object, all that's likely to change is the authorizer, so I also added a line for updating expiry dates, which will probably happen here for proxy.